webapp/views/App/views/Home/Dashboard/ActivityLog/store/actions/useGetActivityLogMessages.js

```diff
@@ -31,12 +38,24 @@ export const useFetchMessages = ({ messages, setMessages }) => {
         data: { activityLogs },
       } = await axios.get(`/api/survey/${surveyId}/activity-log`, { params })
 
-      if (R.isEmpty(activityLogs)) {
+      if (A.isEmpty(activityLogs)) {
         return null
       }
 
+      const [taxonomies, categories] = await Promise.all([
+        API.fetchTaxonomies({ surveyId }),
+        API.fetchCategories({ surveyId }),
+      ])
```

Comment:
fetching all taxonomies and categories doesn't look right to me.
I would rather add the taxonomy and category to the respective item on the server side